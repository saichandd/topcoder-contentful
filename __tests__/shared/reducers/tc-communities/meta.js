import mockDate from 'mockdate';
import { mock } from 'topcoder-react-lib';

const { mockAction } = mock;

const mockMetaActions = {
  tcCommunities: {
    meta: {
      mobileToggle: mockAction('TC_COMMUNITIES/META/MOBILE_TOGGLE'),
      fetchDataInit: mockAction('TC_COMMUNITIES/META/FETCH_DATA_INIT', 'test-community'),
      fetchDataDone: mockAction(
        'TC_COMMUNITIES/META/FETCH_DATA_DONE',
        Promise.resolve('payload'),
      ),
    },
  },
};
jest.setMock(require.resolve('actions/tc-communities/meta'), mockMetaActions);

const reducer = require('reducers/tc-communities/meta');

beforeAll(() => {
  mockDate.set('2018-06-03T02:15:55Z');
});

beforeEach(() => jest.clearAllMocks());

afterAll(() => {
  mockDate.reset();
});

function testReducer(r) {
  let state;

  function check(action) {
    state = r(state, action);
    expect(state).toMatchSnapshot();
    mockDate.set(Date.now() + 1234);
  }

  test('Creates expected intial state', () => {
    check('@@INIT');
  });

  test('Handles fetchDataInit as expected', () => {
    check(mockMetaActions.tcCommunities.meta.fetchDataInit());
  });

  test('Handles mobileToggle as expected', () => {
    check(mockMetaActions.tcCommunities.meta.mobileToggle());
    check(state, mockMetaActions.tcCommunities.meta.mobileToggle());
  });
}

describe('Default reducer', () => {
  testReducer(reducer.default);
});

describe('Factory without http request', () => reducer.factory().then(testReducer));

describe('Factory with server-side rendering', () => reducer.factory({
  url: '/community/communityId/header',
}).then(testReducer));

describe('Factory without server-side rendering', () => reducer.factory({
  url: '/some-random-url',
}).then(testReducer));
