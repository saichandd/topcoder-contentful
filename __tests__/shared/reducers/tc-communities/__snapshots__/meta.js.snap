// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default reducer Creates expected intial state 1`] = `
Object {
  "data": null,
  "loadingOperationId": null,
  "numRefs": 0,
  "timestamp": 0,
}
`;

exports[`Default reducer Handles fetchDataInit as expected 1`] = `
Object {
  "data": null,
  "loadingOperationId": "test-community",
  "numRefs": 0,
  "timestamp": 0,
}
`;

exports[`Default reducer Handles mobileToggle as expected 1`] = `
Object {
  "data": null,
  "isMobileOpen": true,
  "loadingOperationId": "test-community",
  "numRefs": 0,
  "timestamp": 0,
}
`;

exports[`Default reducer Handles mobileToggle as expected 2`] = `
Object {
  "data": null,
  "isMobileOpen": true,
  "loadingOperationId": "test-community",
  "numRefs": 0,
  "timestamp": 0,
}
`;
